### PR TITLE
ceph-osd.sls: use new OSD creation syntax

### PR DESCRIPTION
**Pull request by smithfarm**
References: https://github.com/ceph/ceph-salt/issues/118
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Review by sebastian-philipp — APPROVED** [review]

File: ceph-salt-formula/salt/ceph-salt/ceph-osd.sls [hunk 1]
```diff
@@ -11,7 +11,7 @@ deploy ceph osds ({{ loop.index }}/{{ dg_list | length }}):
   cmd.run:
     - name: |
         ceph orch device ls --refresh
-        echo '{{ dg_spec }}' | ceph orch osd create -i -
+        echo '{{ dg_spec }}' | ceph orch apply osd -i -
```
Comment (ricardoasmarques) [inline ceph-salt-formula/salt/ceph-salt/ceph-osd.sls:14]: I'm afraid we need to adapt the `dg_spec` too:
```
node1:~ # echo ''{"testing_dg_node1": {"host_pattern": "node1*", "data_devices": {"all": true}}}'' | ceph orch apply osd -i -
Error ENOENT: Spec needs a "service_type" key.
```

Comment (smithfarm) [reply ceph-salt-formula/salt/ceph-salt/ceph-osd.sls:14]: hm, `service_type` does not seem to be documented at https://docs.ceph.com/docs/master/cephadm/drivegroups/#drivegroups

@sebastian-philipp ?

Comment (ricardoasmarques) [reply ceph-salt-formula/salt/ceph-salt/ceph-osd.sls:14]: @smithfarm Looking into https://github.com/ceph/ceph/pull/33922 I think this is the new format:
```
{'service_type': 'osd', 'placement': {'host_pattern': 'test'}, 'service_id': 'foo', 'data_devices': {'all': True}}
```

**Comment by smithfarm** [comment]
Merging this now to get a clean deployment test run in https://github.com/SUSE/sesdev/pull/178

File: ceph-salt-formula/salt/ceph-salt/ceph-osd.sls (hunk 1 above)
Comment (smithfarm) [reply ceph-salt-formula/salt/ceph-salt/ceph-osd.sls:14]: @ricardoasmarques OK, I implemented the new syntax at https://github.com/SUSE/sesdev/pull/178 but the deployment smoke test is still not green :-( Please take a look at the log there.

**Pull request merged**